controllers/users.js

```diff
@@ -0,0 +1,299 @@
+const Address = require("../models/address");
+const Images = require("../models/images");
+const Token = require("../models/token");
+const User = require("../models/user");
+const bcrypt = require("bcrypt");
+const cloudinary = require("cloudinary");
+const sendEmail = require("../utils/sendEmail");
+const jwt = require("jsonwebtoken");
+const { send } = require("process");
+const { createTransport } = require("nodemailer");
+require("dotenv").config();
+
+cloudinary.config({
+  cloud_name: process.env.CLOUD_NAME,
+  api_key: process.env.API_KEY,
+  api_secret: process.env.API_SECRET,
+});
+
+const register = async (req, res) => {
+  try {
+    const salt = await bcrypt.genSalt(10);
+    password = req.body.password;
+    con_password = req.body.con_password;
+    if (password == con_password) {
+      let secretPassword = await bcrypt.hash(password, salt);
+      let user = {
+        username: req.body.username,
+        email: req.body.email,
+        password: secretPassword,
+        firstname: req.body.firstname,
+        lastname: req.body.lastname,
+      };
+      let createdUser = await User.create(user);
+      await sendEmail(
+        user.email,
+        "registration",
+        "user regitered successfully"
+      );
+      res.send({
+        error: 0,
+        message: "user created successfully",
+        data: createdUser,
+      });
+    } else {
+      res.send("password don't match");
+    }
+  } catch (err) {
+    res.send({
+      error: 1,
+      message: err.message || "failed to save new user",
+      data: err,
+    });
+  }
+};
+
+const login = async (req, res) => {
+  let user = await User.findOne({ where: { username: req.body.username } });
+  try {
+    if (user) {
+      const validPassword = await bcrypt.compare(
+        req.body.password,
+        user.password
+      );
+      if (validPassword) {
+        let access_token = jwt.sign(
+          {
+            userId: user.id,
+          },
+          process.env.SECRET_KEY,
+          { expiresIn: "1h" }
+        );
+        let token = {
+          userId: user.id,
+          token: access_token,
+        };
+        let createdToken = await Token.create(token);
```

Comment:
on every login you are creating token, 
so how do you manage the garbage token/unused token ?